uberfire-structure/uberfire-structure-backend/src/main/java/org/guvnor/structure/backend/config/ConfigurationServiceImpl.java

```diff
@@ -288,24 +287,27 @@ public void endBatch() {
         // Return the updated cache
         return configGroupsByTypeWithNamespace.get(type);
     }
-
     private List<ConfigGroup> getConfiguration(final Path dir,
-                                               final ConfigType type) {
+                                               final ConfigType type,
+                                               final String namespace) {
         final List<ConfigGroup> configGroups = new ArrayList<>();
 
         if (!ioService.exists(dir)) {
             return configGroups;
         }
 
         final DirectoryStream<Path> foundConfigs = getDirectoryStreamForFilesWithParticularExtension(dir,
-                                                                                                     type.getExt());
+                type.getExt());
 
         //Only load and cache if a file was found!
         final Iterator<Path> it = foundConfigs.iterator();
         if (it.hasNext()) {
             while (it.hasNext()) {
                 final String content = ioService.readAllString(it.next());
                 final ConfigGroup configGroup = marshaller.unmarshall(content);
+                if (namespace != null) {
```

Comment:
In this case, what happens is there is no namespace (or it is null)?

Reply:
@adrielparedes ideally it should fetch the namespace from the repository config file, as a result of unmarshalling on line 307. But we found that in some versions, config file didn't contain the namespace. so if the namespace is null on line 308, this means either the namespace is fetched from the file or the resource doesn't require a namespace at all (like spaces).